frontend/benefit/applicant/src/constants.ts

```diff
@@ -1,5 +1,11 @@
 export const IS_CLIENT = typeof window !== 'undefined';
 
+export enum ROUTES {
```

Comment:
You can actually combine the locale and the route, for example:
http://localhost:3000/fi/application/new
http://localhost:3000/sv/application/new
Seem to work correctly out of the box.
On the other hand The problem in the language selector in the header is that it does not change the locale path parameter, so page refresh always defaults back to finnish,

Reply:
You still have to save the selected language to the application state somehow if you want to preserve it for page refresh. If it's not in the url, then perhaps redux+localstorage is another option? I would prefer the url.